enclave-manager/web/src/emui/catalog/Catalog.tsx

```diff
@@ -1,33 +1,150 @@
-import { Box, Flex } from "@chakra-ui/react";
-import { Suspense } from "react";
-import { Await, useLoaderData } from "react-router-dom";
+import { SmallCloseIcon } from "@chakra-ui/icons";
+import {
+  Box,
+  Card,
+  CardBody,
+  CardHeader,
+  Flex,
+  Heading,
+  Icon,
+  IconButton,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  InputRightElement,
+  Text,
+} from "@chakra-ui/react";
+import { useEffect, useMemo, useRef, useState } from "react";
+import { FiSearch } from "react-icons/fi";
+import { MdBookmarkAdded } from "react-icons/md";
+import { GetPackagesResponse, KurtosisPackage } from "../../client/packageIndexer/api/kurtosis_package_indexer_pb";
+import { AppPageLayout } from "../../components/AppLayout";
+import { KurtosisPackageCardGrid } from "../../components/catalog/KurtosisPackageCardGrid";
+import { OmniboxCommand } from "../../components/KeyboardCommands";
 import { KurtosisAlert } from "../../components/KurtosisAlert";
-import { CatalogLoaderResolved } from "./loader";
+import { PageTitle } from "../../components/PageTitle";
+import { useCatalogContext } from "./CatalogContext";
 
 export const Catalog = () => {
-  const { catalog } = useLoaderData() as CatalogLoaderResolved;
+  const { catalog, savedPackages } = useCatalogContext();
 
-  return (
-    <Suspense>
-      <Await resolve={catalog} children={(catalog) => <CatalogImpl catalog={catalog} />} />
-    </Suspense>
-  );
+  if (catalog.isErr) {
+    return (
+      <AppPageLayout>
+        <KurtosisAlert message={catalog.error} />
+      </AppPageLayout>
+    );
+  }
+
+  return <CatalogImpl savedPackages={savedPackages} catalog={catalog.value} />;
 };
 
 type CatalogImplProps = {
-  catalog: CatalogLoaderResolved["catalog"];
+  catalog: GetPackagesResponse;
+  savedPackages: KurtosisPackage[];
 };
 
-const CatalogImpl = ({ catalog }: CatalogImplProps) => {
-  if (catalog.isErr) {
-    return <KurtosisAlert message={catalog.error} />;
-  }
+const CatalogImpl = ({ catalog, savedPackages }: CatalogImplProps) => {
+  const searchRef = useRef<HTMLInputElement>(null);
+  const [searchTerm, setSearchTerm] = useState("");
+  const isSearching = searchTerm.length > 0;
+  const filteredCatalog = useMemo(
+    () => catalog.packages.filter((kurtosisPackage) => kurtosisPackage.name.toLowerCase().indexOf(searchTerm) > 0),
+    [searchTerm, catalog],
+  );
+
+  useEffect(() => {
+    const listener = function (e: KeyboardEvent) {
+      const element = searchRef?.current;
+      if ((e.ctrlKey && e.keyCode === 75) || (e.metaKey && e.keyCode === 75)) {
```

Comment:
@Dartoxian wondering if we can centralize this code so we have one place where we define how to hit ctrl/command F?

Reply:
yep good point - we just want a `useCtrlKeyListener` type of hook. I'll add